editor/src/components/editor/actions/actions.tsx

```diff
@@ -5527,6 +5528,32 @@ export const UPDATE_FNS = {
       }
     }
   },
+  UPDATE_CODE_FROM_COLLABORATION_UPDATE: (
+    action: UpdateCodeFromCollaborationUpdate,
+    editor: EditorModel,
+    serverState: ProjectServerState,
+    dispatch: EditorDispatch,
+    builtInDependencies: BuiltInDependencies,
+  ): EditorModel => {
+    // When the current user does own the project...
+    if (serverState.isMyProject === 'yes') {
+      // ...Disallow these updates as they're coming from non-owners.
+      return editor
```

Comment:
imho at this point this would deserve its own thing, or maybe even treat "editor only" updates as a middleware